### PR TITLE
Pins are normalised when the labels are string

### PR DESCRIPTION
**Pull request by imrishabh18**
Pins are normalised for both the string and number scenario

### CODE REVIEW

**Review by seveibar — CHANGES_REQUESTED** [review]
we should not change tests to skip tests to avoid regressions

**Comment by seveibar** [comment]
@imrishabh18 the only time you can do skip tests is when they're first introduced- when this code is merged it's going to break snippets. I'm also not really sure what this code does yet for schematic port arrangements, it seems like it's going to break it

**Comment by imrishabh18** [comment]
@seveibar I will push the update which fixes pins for numbers, but then that will make the PR huge again.

**Review by seveibar — CHANGES_REQUESTED** [review]
The code is just too complex. That giant ternary statement.... its too hard to tell what ot does already and of we build on it itll become unmaintainable

File: lib/utils/schematic/getAllDimensionsForSchematicBox.ts [hunk 1]
```diff
@@ -185,7 +186,8 @@ export const getAllDimensionsForSchematicBox = (
   for (let sideIndex = 0; sideIndex < sidePinCounts.bottomSize; sideIndex++) {
     const pinNumber =
       params.schPortArrangement &&
-      isExplicitPinMappingArrangement(params.schPortArrangement)
+      isExplicitPinMappingArrangement(params.schPortArrangement) &&
+      typeof params.schPortArrangement.bottomSide?.pins[sideIndex] !== "string"
```
Comment (seveibar) [inline lib/utils/schematic/getAllDimensionsForSchematicBox.ts:190]: `"pin1": string` === `1:number`

**Pull request closed without merging**